### ISSUE TITLE
Error response from daemon: dial unix connect: connection refused

### ISSUE THREAD

**Issue by bri365** [docker/for-mac#26]
### Expected behavior
Docker keeps running the container

### Actual behavior
Docker drops connection

### Information
shortly after starting, docker loses connection to my container; worked pretty reliably until RC4 - did a timeout get lowered? 

Diagnostic ID: 31B424D2-BFB3-47AF-AEB3-D8A375553650
Docker for Mac: 1.12.0 (Build 10871)
macOS: Version 10.11.6 (Build 15G31)
[OK]     docker-cli
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

1. build container with postgresql, nodejs, nginx, salt, python, rabbitmq
2. start and initialize...
[20160729-183534.tar.gz](https://github.com/docker/for-mac/files/391642/20160729-183534.tar.gz)
[20160729-183602.tar.gz](https://github.com/docker/for-mac/files/391643/20160729-183602.tar.gz)

**Comment by samoht** [comment]
@djs55 seems that `slirp` is dying:

```
Jul 29 18:35:34 briMac5 Docker[com.docker.vmnetd][5856] <Notice>: EOF reading packet from Unix domain socket: closing
Jul 29 18:35:34 briMac5 Docker[com.docker.vmnetd][5856] <Critical>: Failed to read hello from client
Jul 29 18:35:34 briMac5 Docker[com.docker.slirp][5809] <Error>: failed to establish 9P connection: Caught EOF on underlying FLOW
 ```

**Comment by djs55** [comment]
@samoht I think those errors are caused by the Unix domain socket probing done by `diagnose`.

**Comment by djs55** [comment]
We need a more detailed reproduction case to make progress with this. Could you supply a simple `docker-compose.yml` or similar? When you say "docker loses connection to my container" are you thinking of `docker exec` (or similar) or a network connection with `ssh`?

There are a lot of bug fixes in docker 1.12.1 which is already available in the beta channel. Hopefully a stable update will be released later this week.

**Comment by dave-tucker** [comment]
Closing as this hasn't been updated in a while.
Please try the latest stable version , 1.12.1 and let us know if the issue persists.
